src/imports.rs

```diff
@@ -0,0 +1,127 @@
+/// Identifies the slot names of the dataset item to import the annotation into
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct AnnotationContext {
+    /// List of slot IDs in the dataset item in V7 to attach the annotations to
+    /// Although this field is called slot names, V7 actually expects slot IDs to be provided
+    pub slot_names: Vec<String>,
+}
+
+/// Struct representing an annotation payload data of a V7 annotation suitable for importing back into a V7 dataset item
+/// One instance of this struct corresponds to one annotation being imported into the dataset item
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct AnnotationImportAnnotation {
+    pub id: String,
+    pub data: AnnotationImportData,
+    pub annotation_class_id: u32,
+    pub context_keys: AnnotationContext,
+}
+
+/// Struct representing a complete annotation import payload of V7 annotations into a single V7 dataset item
+/// There will be one instance of this struct for each dataset item, where each struct contains many items for the annotations its importing
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct AnnotationImport {
+    pub annotations: Vec<AnnotationImportAnnotation>,
+    pub overwrite: bool,
+}
+
+impl From<Vec<Keypoint>> for AnnotationImportPolygon {
+    fn from(value: Vec<Keypoint>) -> Self {
+        AnnotationImportPolygon { path: value }
+    }
+}
+
+impl From<Vec<Keypoint>> for AnnotationImportData {
+    fn from(value: Vec<Keypoint>) -> Self {
+        AnnotationImportData {
+            polygon: Some(AnnotationImportPolygon::from(value)),
+            tag: None,
+        }
+    }
+}
+
+fn _find_annotation_class_id(
+    eligible_annotation_classes: &[&AnnotationClass],
+    class_name: &str,
+) -> Result<u32> {
+    eligible_annotation_classes
+        .iter()
+        .find(|ac| {
+            if ac.name.is_some() {
+                ac.name.clone().unwrap() == class_name
+            } else {
+                false
+            }
+        })
+        .context("Unable to find matching annotation class ID from export JSON")?
+        .id
+        .context("Annotation Class has no ID")
+}
+
+impl AnnotationImportAnnotation {
+    pub fn new_polygon_annotation(
+        original_annotation: &ImageAnnotation,
+        path: Vec<Keypoint>,
+        eligible_annotation_classes: &[&AnnotationClass],
+        slot_name: &str,
+    ) -> Result<Self> {
+        Ok(AnnotationImportAnnotation {
+            id: uuid::Uuid::new_v4().to_string(),
+            data: AnnotationImportData::from(path),
+            annotation_class_id: _find_annotation_class_id(
+                eligible_annotation_classes,
+                &original_annotation.name,
+            )?,
+            context_keys: AnnotationContext {
+                slot_names: vec![slot_name.to_string()],
+            },
+        })
+    }
+
+    pub fn new_tag_annotation(
```

Comment:
**thought**: Similarly for these methods in the impl, these sound to be datalake specific? Are there some unit tests to know how these could be used?

Reply:
I did consider placing these functionalities in the datalake, but ultimately decided to integrate them into the Darwin Rust client for a couple of key reasons. These methods are primarily constructors and builders for the structs within the client. Keeping them in the client ensures that the code remains cohesive and easy to manage. It also simplifies the user experience, as developers can directly use and understand these structures within a single, unified context.

Reply:
Ah I see, that makes sense!

Do you reckon it would be useful to have a comment or a docstring or a unit test on how these `new_<..>_annotation` methods can be used ?